build_tools/cmake/iree_lit_test.cmake

```diff
@@ -80,8 +80,11 @@ function(iree_lit_test)
   list(APPEND _RULE_LABELS "${_PACKAGE_PATH}")
   set_property(TEST ${_NAME_PATH} PROPERTY LABELS "${_RULE_LABELS}")
   set_property(TEST ${_NAME_PATH} PROPERTY REQUIRED_FILES "${_TEST_FILE_PATH}")
+  set(_TEST_TMPDIR "${IREE_TEST_TMPDIR}/${_NAME_PATH}_test_tmpdir")
```

Comment:
For consistency, we probably also want to label these tests with "requires-tmpdir" in bazel and cmake, if we use that tag at all.